ChatTTS/core.py

```diff
@@ -519,65 +530,23 @@ def _infer_code(
             )
             input_ids = [i.tolist() for i in input_ids]
 
-            result = gpt.llm.generate(
-                None,
-                sample_params,
-                input_ids,
+            results_generator = gpt.llm.llm_engine.generate(
+                None, sample_params, uuid.uuid4(), speaker_embedding_param, input_ids[0]
             )
-
-            token_ids = []
-            hidden_states = []
-            for i in result:
-                token_ids.append(torch.tensor(i.outputs[0].token_ids))
-                hidden_states.append(
-                    i.outputs[0].hidden_states.to(torch.float32).to(self.device)
-                )
-
-            del text_mask, input_ids
-
-            return [
-                GPT.GenerationOutputs(
-                    ids=token_ids,
-                    hiddens=hidden_states,
-                    attentions=[],
-                ),
-            ]
-
-        emb = self.embed(input_ids, text_mask)
-
-        del text_mask
-
-        if params.spk_emb is not None:
-            self.speaker.apply(
-                emb,
-                params.spk_emb,
-                input_ids,
-                self.tokenizer.spk_emb_ids,
-                self.gpt.device_gpt,
-            )
-
-        result = gpt.generate(
```

Comment:
Please DO NOT remove the non-vLLM part. The vLLM is always an alternative speed up option since it cannot be used in EVERY devices.

Reply:
ok